easybuild/easyblocks/generic/configuremake.py

```diff
@@ -83,13 +88,35 @@ def configure_step(self, cmd_prefix=''):
         if prefix_opt is None:
             prefix_opt = '--prefix='
 
-        cmd = "%(preconfigopts)s %(cmd_prefix)s./configure %(prefix_opt)s%(installdir)s %(configopts)s" % {
-            'preconfigopts': self.cfg['preconfigopts'],
-            'cmd_prefix': cmd_prefix,
-            'prefix_opt': prefix_opt,
-            'installdir': self.installdir,
-            'configopts': self.cfg['configopts'],
-        }
+        configure_command = cmd_prefix + './configure'
+
+        # Avoid using config.guess from an Autoconf generated package as it is frequently out of date
+        # use the version shipped with EB instead and provide the result to the configure command
+        build_type_option = ''
+        # possible that the configure_command is generated using preconfigopts...we're at the mercy of the gods then
+        if os.path.isfile(configure_command):
+            if 'Generated by GNU Autoconf' in read_file(configure_command):
+                build_type = self.cfg.get('build_type')
+
+                if build_type is None:
+                    config_guess_path = which('config.guess')
+                    if config_guess_path is None:
+                        self.log.info("No config.guess in $PATH, not setting --build")
```

Comment:
Fatal is too much I think since it is (in most cases) not really required. The whole problem being solved here only really appears for (at present) ARM/Power9

Reply:
I'll upgrade it to a `print_warning`

Reply:
I understand that, but if `config.guess` can't be found something is wrong with how the easyblocks were installed, and people should know about it?

For people that simply `git clone` the repo and use the easyblocks from there, they should be aware that we're shipping a script too now that should be added to `$PATH`?

Reply:
I've made a more explicit `print_warning` there now